### PR TITLE
Add a `libarrow-dev` package to setup script

### PR DESCRIPTION
**Pull request by otegami**
## What I did
- Add `libarrow-dev` package to setup script

## The Error I faced
- I faced the following CMake Error when I build groonga with `--preset=doc`

```console
% cmake -S . -B ../groonga.doc --preset=doc
CMake Error at CMakeLists.txt:531 (find_package):
  By not providing "FindArrow.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Arrow", but
  CMake did not find one.

  Could not find a package configuration file provided by "Arrow" with any of
  the following names:

    ArrowConfig.cmake
    arrow-config.cmake

  Add the installation prefix of "Arrow" to CMAKE_PREFIX_PATH or set
  "Arrow_DIR" to a directory containing one of the above files.  If "Arrow"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Configuring incomplete, errors occurred!
See also "/home/otegami/dev/groonga.doc/CMakeFiles/CMakeOutput.log".
See also "/home/otegami/dev/groonga.doc/CMakeFiles/CMakeError.log".
cmake -S . -B ../groonga.doc --preset=doc  2.91s user 1.14s system 99% cpu 4.066 total
```

### The cause of this error
- There is no step to install Apache Arrow although we need to use it for generating documentation.

### How to reproduce
```console
% cd groonga
% ./setup.sh
% cmake -S . -B ../groonga.doc --preset=doc
-- AVX2 flags: -mavx2
CMake Error at CMakeLists.txt:531 (find_package):
  By not providing "FindArrow.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Arrow", but
  CMake did not find one.

  Could not find a package configuration file provided by "Arrow" with any of
  the following names:

    ArrowConfig.cmake
    arrow-config.cmake

  Add the installation prefix of "Arrow" to CMAKE_PREFIX_PATH or set
  "Arrow_DIR" to a directory containing one of the above files.  If "Arrow"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Configuring incomplete, errors occurred!
See also "/home/otegami/dev/groonga.doc/CMakeFiles/CMakeOutput.log".
See also "/home/otegami/dev/groonga.doc/CMakeFiles/CMakeError.log".
```

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -64,3 +69,6 @@ case "${distribution}-${code_name}" in
     ;;
 esac
 
+if [ -f "apache-arrow-apt-source-latest-${code_name}.deb" ]; then
+  rm "apache-arrow-apt-source-latest-${code_name}.deb"
+fi
```
Comment (otegami) [inline setup.sh:74]: Could you give me some advice about how to handle with this?

Comment (kou) [inline setup.sh:74]: Why do you remove this here instead of after L31?

Comment (otegami) [reply setup.sh]: FIx: [e62ae2f](https://github.com/groonga/groonga/pull/1710/commits/e62ae2ff54c9267aeefec2906f6c7feaa707d25d)

**Comment by otegami** [comment]
@kou 
Thank you for reviewing. I've fixed what you pointed.

File: setup.sh [hunk 2]
```diff
@@ -26,6 +26,11 @@ fi
 if [ -f /etc/debian_version ]; then
   ${SUDO} apt update
   ${SUDO} apt install -y -V lsb-release wget
+
+  wget https://apache.jfrog.io/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
+  ${SUDO} apt install -y -V ./apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
+  rm "apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb"
```
Comment (kou) [inline setup.sh:32]:
```suggestion
  rm apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
```

Comment (otegami) [reply setup.sh]: fix: [1eecc96](https://github.com/groonga/groonga/pull/1710/commits/1eecc96dda2ea44d8f5a7830136d77e89848c5b9)

**Pull request merged**